### PR TITLE
meep: fixing mpi runtime

### PR DESCRIPTION
**Pull request by sheepforce**
@FabianGD just discovered runtime problems with MPI parallel Meep execution. This affects writing HDF5-files mainly. This PR fixes this.

### CODE REVIEW

File: pkgs/apps/meep/default.nix [hunk 1]
```diff
@@ -71,27 +71,41 @@ buildPythonPackage rec {
...
-    sim = mp.Simulation(cell_size=cell,
-                    boundary_layers=pml_layers,
-                    geometry=geometry,
-                    sources=sources,
-                    resolution=resolution)
-    sim.run(until=200)
-    EOF
-  '';
+  checkPhase =
+    let testInput = writeTextFile {
```
Comment (markuskowa) [inline pkgs/apps/meep/default.nix:75]: I would rather keep the
```
cat  > test.py << EOF
...
EOF
```
construction here. That should be faster than building the text file.

File: pkgs/apps/meep/default.nix [hunk 2]
```diff
@@ -71,27 +71,41 @@ buildPythonPackage rec {
...
+    in ''
+      export PYTHONPATH="$out/lib/${python.libPrefix}/site-packages:$PYTHONPATH"
+
+      export OMP_NUM_THREADS=1
+
+      # Fix to make mpich run in a sandbox
+      export HYDRA_IFACE=lo
+      export OMPI_MCA_rmaps_base_oversubscribe=1
+
+      cat ${testInput}
```
Comment (markuskowa) [inline pkgs/apps/meep/default.nix:105]: That was for debugging purposes?

File: pkgs/apps/meep/default.nix (hunk 1 above)
Comment (sheepforce) [reply pkgs/apps/meep/default.nix:75]: OK, can change it again. I've removed the original version as only one of the launched interpreters got the input. But of course we can use cat as you suggest.

File: pkgs/apps/meep/default.nix (hunk 2 above)
Comment (sheepforce) [reply pkgs/apps/meep/default.nix:105]: It was. missed it again, sorry.

**Review by markuskowa — APPROVED** [review]

**Pull request merged**